From 3ab285c9dd29095ecd52ad22fc8907af717fd553 Mon Sep 17 00:00:00 2001
From: neatnoise <neatnoise@gmail.com>
Date: Sun, 6 Sep 2026 03:12:15 +0800
Subject: [PATCH 5/6] [LIZARDBYTE] Fix vulkan_encode returning success when
 video encode queue is missing

When ff_vk_qf_find() fails to find a video encode queue, the error path
returns the stale 'err' variable which is 0 from the previous successful
ff_vk_load_functions() call. This causes avcodec_open2() to report success
despite the encoder being in a broken state, leading to a SIGSEGV when
the caller attempts to encode a frame.

[ https://github.com/LizardByte/build-deps/pull/660 ]

Signed-off-by: Lain "Fearyncess" Yang <i@lain.vg>
---
 libavcodec/vulkan_encode.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavcodec/vulkan_encode.c b/libavcodec/vulkan_encode.c
index 39c43300c4..8f50963bf1 100644
--- a/libavcodec/vulkan_encode.c
+++ b/libavcodec/vulkan_encode.c
@@ -813,7 +813,7 @@ av_cold int ff_vulkan_encode_init(AVCodecContext *avctx, FFVulkanEncodeContext *
     if (!ctx->qf_enc) {
         av_log(avctx, AV_LOG_ERROR, "Encoding of %s is not supported by this device\n",
                avcodec_get_name(avctx->codec_id));
-        return err;
+        return AVERROR(ENOSYS);
     }
 
     /* Load all properties */
-- 
2.55.0

